cmd/meroxa/root/pipelines/create_test.go

```diff
@@ -222,4 +242,56 @@ Pipeline %q successfully created!
 	if !reflect.DeepEqual(gotPipeline, *p) {
 		t.Fatalf("expected \"%v\", got \"%v\"", *p, gotPipeline)
 	}
+
+	// Clear environments feature flags
+	global.Config.Set(global.UserFeatureFlagsEnv, startingFlags)
+}
+
+func TestCreatePipelineWithEnvironmentExecutionWithoutFeatureFlag(t *testing.T) {
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	client := mock.NewMockClient(ctrl)
+	logger := log.NewTestLogger()
+	pName := "my-pipeline"
+	env := "my-env"
+
+	c := &Create{
+		client: client,
+		logger: logger,
+	}
+
+	pi := &meroxa.CreatePipelineInput{
+		Name:        pName,
+		Environment: &meroxa.EnvironmentIdentifier{Name: env},
+	}
+
+	p := &meroxa.Pipeline{
+		ID:   1,
+		Name: pName,
+		Environment: &meroxa.EnvironmentIdentifier{
+			UUID: "2560fbcc-b9ee-461a-a959-fa5656422dc2",
+			Name: env,
+		},
+		State: "healthy",
+	}
+
+	p.Name = pName
+
+	c.args.Name = pi.Name
+	c.flags.Environment = pi.Environment.Name
+
+	err := c.Execute(ctx)
+
+	if err == nil {
+		t.Fatalf("unexpected success")
+	}
+
+	gotError := err.Error()
+	wantError := `no access to the Meroxa self-hosted environments feature.
+Sign up for the Beta here: https://share.hsforms.com/1Uq6UYoL8Q6eV5QzSiyIQkAc2sme`
```

Comment:
Unrelated to this PR, but I wonder if we could use a URL shortener and looked more "meroxa". I believe it's actually possible to have even an URL shortener using our own domain. Not 100% sure though.